cpp/modmesh/view/RPythonConsoleDockWidget.cpp

```diff
@@ -112,13 +125,101 @@ void RPythonConsoleDockWidget::executeCommand()
 {
     std::string const code = m_command_edit->toPlainText().toStdString();
     appendPastCommand(code);
-    m_history_edit->insertPlainText(m_command_edit->toPlainText());
-    m_history_edit->insertPlainText("\n");
+    printCommandHistory();
     m_command_edit->setPlainText("");
     m_command_string = "";
     m_current_command_index = static_cast<int>(m_past_command_strings.size());
     auto & interp = modmesh::python::Interpreter::instance();
-    interp.exec_code(code);
+
+    int pipe_stdout_fd[2];
+    int pipe_stderr_fd[2];
+
+#ifdef _MSC_VER
+    /* Open a set of pipes */
+    if (_pipe(pipe_stdout_fd, 256, _O_BINARY) == -1)
+    {
+        throw "failed to open pipe";
+    }
+    if (_pipe(pipe_stderr_fd, 256, _O_BINARY) == -1)
+    {
+        throw "failed to open pipe";
+    }
+#endif
+
+    interp.exec_code(code, pipe_stdout_fd[0], pipe_stderr_fd[0]);
+
+    printCommandOutput(pipe_stdout_fd[1], pipe_stderr_fd[1]);
+    _close(pipe_stdout_fd[0]);
+    _close(pipe_stdout_fd[1]);
+    _close(pipe_stderr_fd[0]);
+    _close(pipe_stderr_fd[1]);
+}
+
+void RPythonConsoleDockWidget::printCommandOutput(int pipe_stdout_out_fd, int pipe_stderr_out_fd)
+{
+    std::string stdout_str;
+    std::string stderr_str;
+
+    {
+        Formatter formatter;
+        char buffer[256];
+        int number_read = 1;
+        while (number_read)
+        {
+            number_read = _read(pipe_stdout_out_fd,
+                             buffer,
+                             256);
+            formatter << buffer << "<br>";
+        }
+        stdout_str = formatter.str();
+    }
+
+    {
+        Formatter formatter;
+        char buffer[256];
+        int number_read;
+        while (number_read)
+        {
+            number_read = _read(pipe_stderr_out_fd,
+                             buffer,
+                             256);
+            formatter << buffer << "<br>";
+        }
+        stderr_str = formatter.str();
+    }
+
+    printCommandStdout(stdout_str);
+    printCommandStderr(stderr_str);
+}
+
+void RPythonConsoleDockWidget::printCommandHistory()
+{
+    QTextCursor cursor = m_history_edit->textCursor();
+    cursor.movePosition(QTextCursor::End);
+    m_history_edit->setTextCursor(cursor);
+    m_history_edit->insertHtml(m_commandHtml + m_command_edit->toPlainText() + m_endHtml);
+    cursor.movePosition(QTextCursor::End);
+    m_history_edit->setTextCursor(cursor);
+}
+
+void RPythonConsoleDockWidget::printCommandStdout(const std::string & stdout_message)
+{
+    QTextCursor cursor = m_history_edit->textCursor();
+    cursor.movePosition(QTextCursor::End);
+    m_history_edit->setTextCursor(cursor);
+    m_history_edit->insertHtml(m_stdoutHtml + QString::fromStdString(stdout_message) + m_endHtml);
```

Comment:
```suggestion
    m_history_edit->insertHtml(m_stdoutHtml + QString::fromStdString(stdout_message).toHtmlEscaped() + m_endHtml);
```

Reply:
I think this is good!  But perhaps for simple coloring, we may also consider to directly use Qt API for changing color?  I am curious which way works better for the terminal.

Reply:
@tigercosmos I am OK to use HTML there for now, but convenience should not be the major factor for deciding which way to go.  The more indirect the longer runtime it takes.  To some point we will need to evaluate and pick a "better" way, of which the metrics are not defined at the moment.